### PR TITLE
One tag

### PR DESCRIPTION
**Pull request by gferraro**
Added tagging logic, so that each track will have one "master" tag.

For wallaby devices to be detected the api needs to return DeviceId and probably GroupId (incase we want to use it) under processing attributes

### CODE REVIEW

File: test_tag_order.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+    assert master_tag[1]["tag"] == "cat"
+
+    # if resent is unidentified use retrained
+    resnet_result[1]["tag"] = "unidentified"
+    master_tag = thermal.get_master_tag(results, wallaby_device=False)
+    assert master_tag[0].model_config.name == "retrained"
+    assert master_tag[1]["tag"] == "cat"
+
+    # if all models are unidentified use unidentified
+    retrained_result[1]["tag"] = "unidentified"
```
Comment (clare) [inline test_tag_order.py:153]: You have tested that all unidentified models returns unidentfied but not that only one unidentified returns unidentified.

File: test_tag_order.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    original_result[1]["tag"] = "unidentified"
+    master_tag = thermal.get_master_tag(results, wallaby_device=False)
+    assert master_tag[1]["tag"] == "unidentified"
+
```
Comment (clare) [inline test_tag_order.py:157]: Missing test for using original if it is the only one that identifies stuff.

**Review by clare — APPROVED** [review]
Looks good thanks GP!  Love the tests :)

File: processing_TEMPLATE.yaml [hunk 3]
```diff
@@ -26,13 +26,38 @@ thermal:
...
+      model_file: "/path/to/retrained"
+      preview: "classified"
+      tag_scores:
+        default: 2
+    - name: "Resnet"
+      model_file: "/path/to/resnet"
+      preview: "classified"
+      tag_scores:
+        default: 3
+    - name: "wallaby"
```
[the remark this answers is not in the document]
Comment (gferraro) [reply processing_TEMPLATE.yaml:52]: yep thanks

File: processing/thermal.py [hunk 4]
```diff
@@ -193,28 +206,99 @@ def upload_tracks(api, recording, main_model, model_results, logger):
...
+def add_track_tags(api, recording, track, model, logger, model_name=None):
+    if model_name is None:
+        model_name = model.model_config.name
+    track_data = {"name": model_name, "algorithmId": model.algorithm_id}
     track_data["all_class_confidences"] = track.get("all_class_confidences")
     predictions = track.get("predictions")
     if predictions:
         track_data["predictions"] = predictions
 
     if track and "tag" in track:
```
[the remark this answers is not in the document]
Comment (gferraro) [reply processing/thermal.py:278]: good point

**Pull request merged**